### DISCUSSION TITLE
How to set back to default material variant?

### DISCUSSION THREAD

**Discussion by Readerr57** [google/model-viewer#3840] [General]
I am trying to reset material variant example to default variant with button. I have same drop down menu that can be found in example. Is there one line command to switch model variant to default? Like this: "model.setVariant = "default";"?

_https://modelviewer.dev/examples/scenegraph/#variants_

**Comment by elalish** [comment]
It's in the last line of that example: `modelviewer.variantName = null` resets to the default variant.

**Comment by Readerr57** [comment]
Thank you for your answer!

I got it and now it is working better. There is still a problem. When I reset variant to default, it goes to default variant, but after 1-2 seconds it flashes back to variant selected earlier. I change main model also and my idea is to go back to default material variant after model has changed. Drop down menu is working normally, it selects default variant normally, only 3D-model flashes back to non default variant.

Something in this code is swapping variant back to earlier selected variant after seconds. I have also emptied drop down menu, but somewhere in the memory is this selected variant still. 

```
<script>
		const modelViewerVariants = document.querySelector("model-viewer#shoe");
		const select = document.querySelector('#variant');

		modelViewerVariants.addEventListener('load', () => {

		modelViewerVariants.variantName = null;

  		const names = modelViewerVariants.availableVariants;
  			for (const name of names) {
    	const option = document.createElement('option');
    		option.value = name;
    		option.textContent = name;
    		select.appendChild(option);
  		}
		});

	select.addEventListener('input', (event) => {
  		modelViewerVariants.variantName = event.target.value === 'default' ? null : event.target.value;
	});
	</script>
```
